api/endpoints/internal.py

```
@@ -204,6 +216,27 @@ def process_material(
+    """
+    logger = logging.getLogger(__name__)
+    try:
+        embeddings = embed_texts_batched(request.texts)
+        if INTERNAL_REQUESTS:
+            INTERNAL_REQUESTS.labels(status="batch_embed_success").inc()
+        return {"embeddings": embeddings}
+    except Exception as e:
+        logger.exception("Error generating batch embeddings")
+        if INTERNAL_REQUESTS:
+            INTERNAL_REQUESTS.labels(status="batch_embed_error").inc()
+        raise HTTPException(status_code=500, detail=f"Failed to generate embeddings: {e}")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)